tests/test_optimizer/test_dist_adafactor.py

```diff
@@ -0,0 +1,719 @@
+                # ROW Parallel
+                if shard_spec.sharding_sequence[-1] == "R":
+                    tp_p = _gather(input_=tp_p, dim=0, process_group=tp_group)  # gather
+            else:
+                # TP bias
+                tp_p = _gather(input_=tp_p, dim=-1, process_group=tp_group)  # gather
+
+        else:
+            # No TP bias
+            pass
+        correctness_verify(p.data, tp_p.data, dtype)
+
+
+@parameterize(
+    "test_config",
+    [
+        {
+            "tp_size": 1,
+            "num_microbatches": 4,
+            "zero_stage": 2,
+            "precision": "bf16",
+        },
+        {
+            "tp_size": 2,
+            "num_microbatches": 4,
+            "zero_stage": 2,
+            "precision": "bf16",
+        },
+        {
+            "tp_size": 4,
+            "num_microbatches": 4,
+            "zero_stage": 2,
+            "precision": "bf16",
+        },
+        {
+            "tp_size": 1,
+            "num_microbatches": 4,
+            "zero_stage": 2,
+            "precision": "fp16",
+        },
+        {
+            "tp_size": 2,
+            "num_microbatches": 4,
+            "zero_stage": 2,
+            "precision": "fp16",
+        },
+        {
+            "tp_size": 4,
+            "num_microbatches": 4,
+            "zero_stage": 2,
+            "precision": "fp16",
+        },
+        {
+            "tp_size": 2,
+            "num_microbatches": 4,
+            "zero_stage": 1,
+            "precision": "bf16",
+        },
+        {
+            "tp_size": 4,
+            "pp_size": 1,
+            "num_microbatches": 4,
+            "zero_stage": 0,
+            "precision": "bf16",
+        },
+    ],
+)
+def exam_bert_test(test_config):
+    sub_model_zoo = model_zoo.get_sub_registry("transformers_bert")
+    test_config["use_lazy_init"] = False
+    test_config["pp_size"] = 1  # Do NOT test Pipeline Parallel
+    test_config["initial_scale"] = 2**10  # avoid overflow
+    model_list = [
+        "transformers_bert",
+        "transformers_bert_for_pretraining",
+        "transformers_bert_lm_head_model",
+        "transformers_bert_for_masked_lm",
+        "transformers_bert_for_sequence_classification",
+        "transformers_bert_for_token_classification",
+        "transformers_bert_for_next_sentence",
+        "transformers_bert_for_mcq",
+        "transformers_bert_for_question_answering",
+    ]
+
+    for name, (model_fn, data_gen_fn, output_transform_fn, loss_fn, _) in sub_model_zoo.items():
+        print(f"model name {name} {name in model_list}")
+        if name in model_list:
+            print(f"{name} check start")
+            (
+                org_model,
+                org_optimizer,
+                sharded_model,
+                sharded_optimizer,
+                criterion,
+                booster,
+            ) = build_model_from_hybrid_plugin(model_fn, loss_fn, test_config, Adafactor, DistributedAdaFactor)
```

Comment:
You'd better add a test for low level zero plugin

Reply:
> You'd better add a test for low level zero plugin

Added in https://github.com/hpcaitech/ColossalAI/pull/5484/commits/6303291c4a4e367ee86313a64c727515bc022d60;

Reply:
Test for low level zero plugin on bert